src/server/middleware/middleware.go

```diff
@@ -29,3 +29,36 @@ func WithMiddlewares(handler http.Handler, middlewares ...Middleware) http.Handl
 	}
 	return handler
 }
+
+// Skipper defines a function to skip middleware.
+// Returning true skips processing the middleware.
+type Skipper func(*http.Request) bool
+
+// DefaultSkipper returns false which processes the middleware.
+func DefaultSkipper(*http.Request) bool {
+	return false
+}
+
+// New make a middleware from fn which type is func(w http.ResponseWriter, r *http.Request, next http.Handler)
+func New(fn func(http.ResponseWriter, *http.Request, http.Handler), skippers ...Skipper) func(http.Handler) http.Handler {
```

Comment:
From the purpose of this func, you can split the fn to `f (rw, r), http.Handler`, it's more clear

Reply:
`New` func convert `func(http.ResponseWriter, *http.Request, http.Handler)` to a middleware.